components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/usermgt/APIKeyMgtRemoteUserClientPool.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *  Copyright WSO2 Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway.handlers.security.usermgt;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.pool.BasePoolableObjectFactory;
+import org.apache.commons.pool.ObjectPool;
+import org.apache.commons.pool.impl.StackObjectPool;
+import org.wso2.carbon.apimgt.gateway.internal.ServiceReferenceHolder;
+
+public class APIKeyMgtRemoteUserClientPool {
+    private static final Log log = LogFactory.getLog(APIKeyMgtRemoteUserClientPool.class);
+
+    private static final APIKeyMgtRemoteUserClientPool instance = new APIKeyMgtRemoteUserClientPool();
+
+    private final ObjectPool clientPool;
+    private static int maxIdle;
+
+    private APIKeyMgtRemoteUserClientPool() {
+        String maxIdleClients = ServiceReferenceHolder.getInstance().getAPIManagerConfiguration().getFirstProperty
+                ("APIKeyValidator.ConnectionPool.MaxIdle");
+        String initIdleCapacity = ServiceReferenceHolder.getInstance().getAPIManagerConfiguration().getFirstProperty
+                ("APIKeyValidator.ConnectionPool.InitIdleCapacity");
+        int initIdleCapSize;
+        if (StringUtils.isNotEmpty(maxIdleClients)) {
+            maxIdle = Integer.parseInt(maxIdleClients);
+        } else {
+            maxIdle = 50;
```

Comment:
shall we define these default values is a suitable constants class

Reply:
Not applicable anymore since the master branch has a different codebase.